// gqty/Selection/SelectionManager.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const sha1 = require('@gqty/utils/sha1');
const cachedJSON = require('../Utils/cachedJSON.js');
const selection = require('./selection.js');

function separateSelectionTypes(selections) {
  let querySelections;
  let mutationSelections;
  let subscriptionSelections;
  for (const selection$1 of selections) {
    switch (selection$1.type) {
      case selection.SelectionType.Query: {
        querySelections || (querySelections = []);
        querySelections.push(selection$1);
        break;
      }
      case selection.SelectionType.Mutation: {
        mutationSelections || (mutationSelections = []);
        mutationSelections.push(selection$1);
        break;
      }
      case selection.SelectionType.Subscription: {
        subscriptionSelections || (subscriptionSelections = []);
        subscriptionSelections.push(selection$1);
        break;
      }
    }
  }
  return {
    querySelections,
    mutationSelections,
    subscriptionSelections
  };
}
const selectionsBackupVersion = "v1";
function isSelectionsBackup(selectionsBackup) {
  return Array.isArray(selectionsBackup) && Array.isArray(selectionsBackup[0]) && selectionsBackup[1] === selectionsBackupVersion;
}
let uniqueSelectionId = 0;
function createSelectionManager() {
  const selectionCache = /* @__PURE__ */ new Map();
  const stringsHash = {};
  let restoredBackup;
  function getSerializedVariablesId(variables) {
    const serializedVariables = cachedJSON.serializeVariables(variables);
    let hashId;
    if ((hashId = stringsHash[serializedVariables]) === void 0) {
      hashId = stringsHash[serializedVariables] = sha1.sha1(serializedVariables).slice(0, 5);
      if (restoredBackup)
        restoredBackup[0].push([serializedVariables, hashId]);
    }
    return hashId;
  }
  function restore(backup2) {
    if (!isSelectionsBackup(backup2))
      return;
    restoredBackup = backup2;
    for (const [stringKey, hashIdValue] of backup2[0]) {
      stringsHash[stringKey] = hashIdValue;
    }
  }
  function backup() {
    if (restoredBackup) {
      restoredBackup[0] = [];
      return restoredBackup;
    }
    const backup2 = [[], selectionsBackupVersion];
    for (const serializedVariables in stringsHash) {
      backup2[0].push([serializedVariables, stringsHash[serializedVariables]]);
    }
    return restoredBackup = backup2;
  }
  function getVariableAlias(key, variables, variableTypes) {
    return key + "_" + getSerializedVariablesId(variableTypes) + "_" + getSerializedVariablesId(variables);
  }
  function getSelection({
    key,
    prevSelection,
    args,
    argTypes,
    type,
    unions
  }) {
    let alias;
    let cacheKey = key + "";
    if (args && argTypes) {
      alias = getVariableAlias(key, args, argTypes);
      cacheKey = alias;
    }
    if (prevSelection) {
      cacheKey = prevSelection.pathString + "." + cacheKey;
    }
    if (unions == null ? void 0 : unions.length) {
      cacheKey += ";" + unions.join(";");
    }
    let selection$1 = selectionCache.get(cacheKey);
    if (selection$1 == null) {
      selection$1 = new selection.Selection({
        key,
        prevSelection,
        args,
        argTypes,
        alias,
        type,
        unions,
        id: ++uniqueSelectionId
      });
      selectionCache.set(cacheKey, selection$1);
    } else if (args) {
      selection$1.args = args;
    }
    return selection$1;
  }
  return {
    getSelection,
    restore,
    backup
  };
}

exports.createSelectionManager = createSelectionManager;
exports.separateSelectionTypes = separateSelectionTypes;
