// @gqty/utils/mergeWith.mjs
"use strict";

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_listCacheClear.js
function Pr() {
  this.__data__ = [], this.size = 0;
}
var ot = Pr;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/eq.js
function Sr(t, r) {
  return t === r || t !== t && r !== r;
}
var g = Sr;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_assocIndexOf.js
function wr(t, r) {
  for (var e = t.length; e--; )
    if (g(t[e][0], r))
      return e;
  return -1;
}
var b = wr;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_listCacheDelete.js
var Ir = Array.prototype, Mr = Ir.splice;
function Er(t) {
  var r = this.__data__, e = b(r, t);
  if (e < 0)
    return !1;
  var o = r.length - 1;
  return e == o ? r.pop() : Mr.call(r, e, 1), --this.size, !0;
}
var at = Er;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_listCacheGet.js
function Lr(t) {
  var r = this.__data__, e = b(r, t);
  return e < 0 ? void 0 : r[e][1];
}
var nt = Lr;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_listCacheHas.js
function Dr(t) {
  return b(this.__data__, t) > -1;
}
var it = Dr;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_listCacheSet.js
function Fr(t, r) {
  var e = this.__data__, o = b(e, t);
  return o < 0 ? (++this.size, e.push([t, r])) : e[o][1] = r, this;
}
var ft = Fr;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_ListCache.js
function _(t) {
  var r = -1, e = t == null ? 0 : t.length;
  for (this.clear(); ++r < e; ) {
    var o = t[r];
    this.set(o[0], o[1]);
  }
}
_.prototype.clear = ot;
_.prototype.delete = at;
_.prototype.get = nt;
_.prototype.has = it;
_.prototype.set = ft;
var y = _;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_stackClear.js
function Gr() {
  this.__data__ = new y(), this.size = 0;
}
var pt = Gr;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_stackDelete.js
function Nr(t) {
  var r = this.__data__, e = r.delete(t);
  return this.size = r.size, e;
}
var st = Nr;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_stackGet.js
function Ur(t) {
  return this.__data__.get(t);
}
var ut = Ur;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_stackHas.js
function zr(t) {
  return this.__data__.has(t);
}
var mt = zr;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_freeGlobal.js
var Hr = typeof global == "object" && global && global.Object === Object && global, U = Hr;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_root.js
var Br = typeof self == "object" && self && self.Object === Object && self, Rr = U || Br || Function("return this")(), c = Rr;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_Symbol.js
var Vr = c.Symbol, O = Vr;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_getRawTag.js
var lt = Object.prototype, Kr = lt.hasOwnProperty, qr = lt.toString, E = O ? O.toStringTag : void 0;
function $r(t) {
  var r = Kr.call(t, E), e = t[E];
  try {
    t[E] = void 0;
    var o = !0;
  } catch {
  }
  var n = qr.call(t);
  return o && (r ? t[E] = e : delete t[E]), n;
}
var ct = $r;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_objectToString.js
var Wr = Object.prototype, Jr = Wr.toString;
function Xr(t) {
  return Jr.call(t);
}
var dt = Xr;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_baseGetTag.js
var Yr = "[object Null]", Zr = "[object Undefined]", ht = O ? O.toStringTag : void 0;
function Qr(t) {
  return t == null ? t === void 0 ? Zr : Yr : ht && ht in Object(t) ? ct(t) : dt(t);
}
var x = Qr;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/isObject.js
function kr(t) {
  var r = typeof t;
  return t != null && (r == "object" || r == "function");
}
var u = kr;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/isFunction.js
var te = "[object AsyncFunction]", re = "[object Function]", ee = "[object GeneratorFunction]", oe = "[object Proxy]";
function ae(t) {
  if (!u(t))
    return !1;
  var r = x(t);
  return r == re || r == ee || r == te || r == oe;
}
var T = ae;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_coreJsData.js
var ne = c["__core-js_shared__"], z = ne;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_isMasked.js
var gt = function() {
  var t = /[^.]+$/.exec(z && z.keys && z.keys.IE_PROTO || "");
  return t ? "Symbol(src)_1." + t : "";
}();
function ie(t) {
  return !!gt && gt in t;
}
var bt = ie;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_toSource.js
var fe = Function.prototype, pe = fe.toString;
function se(t) {
  if (t != null) {
    try {
      return pe.call(t);
    } catch {
    }
    try {
      return t + "";
    } catch {
    }
  }
  return "";
}
var yt = se;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_baseIsNative.js
var ue = /[\\^$.*+?()[\]{}|]/g, me = /^\[object .+?Constructor\]$/, le = Function.prototype, ce = Object.prototype, de = le.toString, he = ce.hasOwnProperty, ge = RegExp("^" + de.call(he).replace(ue, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$");
function be(t) {
  if (!u(t) || bt(t))
    return !1;
  var r = T(t) ? ge : me;
  return r.test(yt(t));
}
var xt = be;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_getValue.js
function ye(t, r) {
  return t?.[r];
}
var vt = ye;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_getNative.js
function xe(t, r) {
  var e = vt(t, r);
  return xt(e) ? e : void 0;
}
var j = xe;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_Map.js
var ve = j(c, "Map"), H = ve;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_nativeCreate.js
var _e = j(Object, "create"), h = _e;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_hashClear.js
function Oe() {
  this.__data__ = h ? h(null) : {}, this.size = 0;
}
var _t = Oe;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_hashDelete.js
function Te(t) {
  var r = this.has(t) && delete this.__data__[t];
  return this.size -= r ? 1 : 0, r;
}
var Ot = Te;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_hashGet.js
var je = "__lodash_hash_undefined__", Ae = Object.prototype, Ce = Ae.hasOwnProperty;
function Pe(t) {
  var r = this.__data__;
  if (h) {
    var e = r[t];
    return e === je ? void 0 : e;
  }
  return Ce.call(r, t) ? r[t] : void 0;
}
var Tt = Pe;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_hashHas.js
var Se = Object.prototype, we = Se.hasOwnProperty;
function Ie(t) {
  var r = this.__data__;
  return h ? r[t] !== void 0 : we.call(r, t);
}
var jt = Ie;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_hashSet.js
var Me = "__lodash_hash_undefined__";
function Ee(t, r) {
  var e = this.__data__;
  return this.size += this.has(t) ? 0 : 1, e[t] = h && r === void 0 ? Me : r, this;
}
var At = Ee;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_Hash.js
function A(t) {
  var r = -1, e = t == null ? 0 : t.length;
  for (this.clear(); ++r < e; ) {
    var o = t[r];
    this.set(o[0], o[1]);
  }
}
A.prototype.clear = _t;
A.prototype.delete = Ot;
A.prototype.get = Tt;
A.prototype.has = jt;
A.prototype.set = At;
var Z = A;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_mapCacheClear.js
function Le() {
  this.size = 0, this.__data__ = {
    hash: new Z(),
    map: new (H || y)(),
    string: new Z()
  };
}
var Ct = Le;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_isKeyable.js
function De(t) {
  var r = typeof t;
  return r == "string" || r == "number" || r == "symbol" || r == "boolean" ? t !== "__proto__" : t === null;
}
var Pt = De;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_getMapData.js
function Fe(t, r) {
  var e = t.__data__;
  return Pt(r) ? e[typeof r == "string" ? "string" : "hash"] : e.map;
}
var v = Fe;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_mapCacheDelete.js
function Ge(t) {
  var r = v(this, t).delete(t);
  return this.size -= r ? 1 : 0, r;
}
var St = Ge;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_mapCacheGet.js
function Ne(t) {
  return v(this, t).get(t);
}
var wt = Ne;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_mapCacheHas.js
function Ue(t) {
  return v(this, t).has(t);
}
var It = Ue;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_mapCacheSet.js
function ze(t, r) {
  var e = v(this, t), o = e.size;
  return e.set(t, r), this.size += e.size == o ? 0 : 1, this;
}
var Mt = ze;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_MapCache.js
function C(t) {
  var r = -1, e = t == null ? 0 : t.length;
  for (this.clear(); ++r < e; ) {
    var o = t[r];
    this.set(o[0], o[1]);
  }
}
C.prototype.clear = Ct;
C.prototype.delete = St;
C.prototype.get = wt;
C.prototype.has = It;
C.prototype.set = Mt;
var Et = C;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_stackSet.js
var He = 200;
function Be(t, r) {
  var e = this.__data__;
  if (e instanceof y) {
    var o = e.__data__;
    if (!H || o.length < He - 1)
      return o.push([t, r]), this.size = ++e.size, this;
    e = this.__data__ = new Et(o);
  }
  return e.set(t, r), this.size = e.size, this;
}
var Lt = Be;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_Stack.js
function P(t) {
  var r = this.__data__ = new y(t);
  this.size = r.size;
}
P.prototype.clear = pt;
P.prototype.delete = st;
P.prototype.get = ut;
P.prototype.has = mt;
P.prototype.set = Lt;
var Dt = P;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_defineProperty.js
var Re = function() {
  try {
    var t = j(Object, "defineProperty");
    return t({}, "", {}), t;
  } catch {
  }
}(), S = Re;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_baseAssignValue.js
function Ve(t, r, e) {
  r == "__proto__" && S ? S(t, r, {
    configurable: !0,
    enumerable: !0,
    value: e,
    writable: !0
  }) : t[r] = e;
}
var w = Ve;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_assignMergeValue.js
function Ke(t, r, e) {
  (e !== void 0 && !g(t[r], e) || e === void 0 && !(r in t)) && w(t, r, e);
}
var L = Ke;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_createBaseFor.js
function qe(t) {
  return function(r, e, o) {
    for (var n = -1, i = Object(r), p = o(r), a = p.length; a--; ) {
      var s = p[t ? a : ++n];
      if (e(i[s], s, i) === !1)
        break;
    }
    return r;
  };
}
var Ft = qe;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_baseFor.js
var $e = Ft(), Gt = $e;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_cloneBuffer.js
var Ht = typeof exports == "object" && exports && !exports.nodeType && exports, Nt = Ht && typeof module == "object" && module && !module.nodeType && module, We = Nt && Nt.exports === Ht, Ut = We ? c.Buffer : void 0, zt = Ut ? Ut.allocUnsafe : void 0;
function Je(t, r) {
  if (r)
    return t.slice();
  var e = t.length, o = zt ? zt(e) : new t.constructor(e);
  return t.copy(o), o;
}
var Bt = Je;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_Uint8Array.js
var Xe = c.Uint8Array, Q = Xe;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_cloneArrayBuffer.js
function Ye(t) {
  var r = new t.constructor(t.byteLength);
  return new Q(r).set(new Q(t)), r;
}
var Rt = Ye;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_cloneTypedArray.js
function Ze(t, r) {
  var e = r ? Rt(t.buffer) : t.buffer;
  return new t.constructor(e, t.byteOffset, t.length);
}
var Vt = Ze;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_copyArray.js
function Qe(t, r) {
  var e = -1, o = t.length;
  for (r || (r = Array(o)); ++e < o; )
    r[e] = t[e];
  return r;
}
var Kt = Qe;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_baseCreate.js
var qt = Object.create, ke = function() {
  function t() {
  }
  return function(r) {
    if (!u(r))
      return {};
    if (qt)
      return qt(r);
    t.prototype = r;
    var e = new t();
    return t.prototype = void 0, e;
  };
}(), $t = ke;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_overArg.js
function to(t, r) {
  return function(e) {
    return t(r(e));
  };
}
var Wt = to;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_getPrototype.js
var ro = Wt(Object.getPrototypeOf, Object), B = ro;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_isPrototype.js
var eo = Object.prototype;
function oo(t) {
  var r = t && t.constructor, e = typeof r == "function" && r.prototype || eo;
  return t === e;
}
var R = oo;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_initCloneObject.js
function ao(t) {
  return typeof t.constructor == "function" && !R(t) ? $t(B(t)) : {};
}
var Jt = ao;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/isObjectLike.js
function no(t) {
  return t != null && typeof t == "object";
}
var d = no;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_baseIsArguments.js
var io = "[object Arguments]";
function fo(t) {
  return d(t) && x(t) == io;
}
var k = fo;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/isArguments.js
var Xt = Object.prototype, po = Xt.hasOwnProperty, so = Xt.propertyIsEnumerable, uo = k(function() {
  return arguments;
}()) ? k : function(t) {
  return d(t) && po.call(t, "callee") && !so.call(t, "callee");
}, D = uo;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/isArray.js
var mo = Array.isArray, F = mo;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/isLength.js
var lo = 9007199254740991;
function co(t) {
  return typeof t == "number" && t > -1 && t % 1 == 0 && t <= lo;
}
var V = co;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/isArrayLike.js
function ho(t) {
  return t != null && V(t.length) && !T(t);
}
var I = ho;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/isArrayLikeObject.js
function go(t) {
  return d(t) && I(t);
}
var Yt = go;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/stubFalse.js
function bo() {
  return !1;
}
var Zt = bo;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/isBuffer.js
var tr = typeof exports == "object" && exports && !exports.nodeType && exports, Qt = tr && typeof module == "object" && module && !module.nodeType && module, yo = Qt && Qt.exports === tr, kt = yo ? c.Buffer : void 0, xo = kt ? kt.isBuffer : void 0, vo = xo || Zt, K = vo;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/isPlainObject.js
var _o = "[object Object]", Oo = Function.prototype, To = Object.prototype, rr = Oo.toString, jo = To.hasOwnProperty, Ao = rr.call(Object);
function Co(t) {
  if (!d(t) || x(t) != _o)
    return !1;
  var r = B(t);
  if (r === null)
    return !0;
  var e = jo.call(r, "constructor") && r.constructor;
  return typeof e == "function" && e instanceof e && rr.call(e) == Ao;
}
var er = Co;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_baseIsTypedArray.js
var Po = "[object Arguments]", So = "[object Array]", wo = "[object Boolean]", Io = "[object Date]", Mo = "[object Error]", Eo = "[object Function]", Lo = "[object Map]", Do = "[object Number]", Fo = "[object Object]", Go = "[object RegExp]", No = "[object Set]", Uo = "[object String]", zo = "[object WeakMap]", Ho = "[object ArrayBuffer]", Bo = "[object DataView]", Ro = "[object Float32Array]", Vo = "[object Float64Array]", Ko = "[object Int8Array]", qo = "[object Int16Array]", $o = "[object Int32Array]", Wo = "[object Uint8Array]", Jo = "[object Uint8ClampedArray]", Xo = "[object Uint16Array]", Yo = "[object Uint32Array]", f = {};
f[Ro] = f[Vo] = f[Ko] = f[qo] = f[$o] = f[Wo] = f[Jo] = f[Xo] = f[Yo] = !0;
f[Po] = f[So] = f[Ho] = f[wo] = f[Bo] = f[Io] = f[Mo] = f[Eo] = f[Lo] = f[Do] = f[Fo] = f[Go] = f[No] = f[Uo] = f[zo] = !1;
function Zo(t) {
  return d(t) && V(t.length) && !!f[x(t)];
}
var or = Zo;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_baseUnary.js
function Qo(t) {
  return function(r) {
    return t(r);
  };
}
var ar = Qo;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_nodeUtil.js
var nr = typeof exports == "object" && exports && !exports.nodeType && exports, G = nr && typeof module == "object" && module && !module.nodeType && module, ko = G && G.exports === nr, tt = ko && U.process, ta = function() {
  try {
    var t = G && G.require && G.require("util").types;
    return t || tt && tt.binding && tt.binding("util");
  } catch {
  }
}(), rt = ta;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/isTypedArray.js
var ir = rt && rt.isTypedArray, ra = ir ? ar(ir) : or, q = ra;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_safeGet.js
function ea(t, r) {
  if (!(r === "constructor" && typeof t[r] == "function") && r != "__proto__")
    return t[r];
}
var N = ea;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_assignValue.js
var oa = Object.prototype, aa = oa.hasOwnProperty;
function na(t, r, e) {
  var o = t[r];
  (!(aa.call(t, r) && g(o, e)) || e === void 0 && !(r in t)) && w(t, r, e);
}
var fr = na;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_copyObject.js
function ia(t, r, e, o) {
  var n = !e;
  e || (e = {});
  for (var i = -1, p = r.length; ++i < p; ) {
    var a = r[i], s = o ? o(e[a], t[a], a, e, t) : void 0;
    s === void 0 && (s = t[a]), n ? w(e, a, s) : fr(e, a, s);
  }
  return e;
}
var pr = ia;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_baseTimes.js
function fa(t, r) {
  for (var e = -1, o = Array(t); ++e < t; )
    o[e] = r(e);
  return o;
}
var sr = fa;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_isIndex.js
var pa = 9007199254740991, sa = /^(?:0|[1-9]\d*)$/;
function ua(t, r) {
  var e = typeof t;
  return r = r ?? pa, !!r && (e == "number" || e != "symbol" && sa.test(t)) && t > -1 && t % 1 == 0 && t < r;
}
var $ = ua;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_arrayLikeKeys.js
var ma = Object.prototype, la = ma.hasOwnProperty;
function ca(t, r) {
  var e = F(t), o = !e && D(t), n = !e && !o && K(t), i = !e && !o && !n && q(t), p = e || o || n || i, a = p ? sr(t.length, String) : [], s = a.length;
  for (var l in t)
    (r || la.call(t, l)) && !(p && (l == "length" || n && (l == "offset" || l == "parent") || i && (l == "buffer" || l == "byteLength" || l == "byteOffset") || $(l, s))) && a.push(l);
  return a;
}
var ur = ca;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_nativeKeysIn.js
function da(t) {
  var r = [];
  if (t != null)
    for (var e in Object(t))
      r.push(e);
  return r;
}
var mr = da;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_baseKeysIn.js
var ha = Object.prototype, ga = ha.hasOwnProperty;
function ba(t) {
  if (!u(t))
    return mr(t);
  var r = R(t), e = [];
  for (var o in t)
    o == "constructor" && (r || !ga.call(t, o)) || e.push(o);
  return e;
}
var lr = ba;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/keysIn.js
function ya(t) {
  return I(t) ? ur(t, !0) : lr(t);
}
var W = ya;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/toPlainObject.js
function xa(t) {
  return pr(t, W(t));
}
var cr = xa;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_baseMergeDeep.js
function va(t, r, e, o, n, i, p) {
  var a = N(t, e), s = N(r, e), l = p.get(s);
  if (l) {
    L(t, e, l);
    return;
  }
  var m = i ? i(a, s, e + "", t, r, p) : void 0, M = m === void 0;
  if (M) {
    var X = F(s), Y = !X && K(s), et = !X && !Y && q(s);
    m = s, X || Y || et ? F(a) ? m = a : Yt(a) ? m = Kt(a) : Y ? (M = !1, m = Bt(s, !0)) : et ? (M = !1, m = Vt(s, !0)) : m = [] : er(s) || D(s) ? (m = a, D(a) ? m = cr(a) : (!u(a) || T(a)) && (m = Jt(s))) : M = !1;
  }
  M && (p.set(s, m), n(m, s, o, i, p), p.delete(s)), L(t, e, m);
}
var dr = va;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_baseMerge.js
function hr(t, r, e, o, n) {
  t !== r && Gt(r, function(i, p) {
    if (n || (n = new Dt()), u(i))
      dr(t, r, p, e, hr, o, n);
    else {
      var a = o ? o(N(t, p), i, p + "", t, r, n) : void 0;
      a === void 0 && (a = i), L(t, p, a);
    }
  }, W);
}
var gr = hr;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/identity.js
function _a(t) {
  return t;
}
var J = _a;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_apply.js
function Oa(t, r, e) {
  switch (e.length) {
    case 0:
      return t.call(r);
    case 1:
      return t.call(r, e[0]);
    case 2:
      return t.call(r, e[0], e[1]);
    case 3:
      return t.call(r, e[0], e[1], e[2]);
  }
  return t.apply(r, e);
}
var br = Oa;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_overRest.js
var yr = Math.max;
function Ta(t, r, e) {
  return r = yr(r === void 0 ? t.length - 1 : r, 0), function() {
    for (var o = arguments, n = -1, i = yr(o.length - r, 0), p = Array(i); ++n < i; )
      p[n] = o[r + n];
    n = -1;
    for (var a = Array(r + 1); ++n < r; )
      a[n] = o[n];
    return a[r] = e(p), br(t, this, a);
  };
}
var xr = Ta;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/constant.js
function ja(t) {
  return function() {
    return t;
  };
}
var vr = ja;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_baseSetToString.js
var Aa = S ? function(t, r) {
  return S(t, "toString", {
    configurable: !0,
    enumerable: !1,
    value: vr(r),
    writable: !0
  });
} : J, _r = Aa;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_shortOut.js
var Ca = 800, Pa = 16, Sa = Date.now;
function wa(t) {
  var r = 0, e = 0;
  return function() {
    var o = Sa(), n = Pa - (o - e);
    if (e = o, n > 0) {
      if (++r >= Ca)
        return arguments[0];
    } else
      r = 0;
    return t.apply(void 0, arguments);
  };
}
var Or = wa;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_setToString.js
var Ia = Or(_r), Tr = Ia;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_baseRest.js
function Ma(t, r) {
  return Tr(xr(t, r, J), t + "");
}
var jr = Ma;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_isIterateeCall.js
function Ea(t, r, e) {
  if (!u(e))
    return !1;
  var o = typeof r;
  return (o == "number" ? I(e) && $(r, e.length) : o == "string" && r in e) ? g(e[r], t) : !1;
}
var Ar = Ea;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/_createAssigner.js
function La(t) {
  return jr(function(r, e) {
    var o = -1, n = e.length, i = n > 1 ? e[n - 1] : void 0, p = n > 2 ? e[2] : void 0;
    for (i = t.length > 3 && typeof i == "function" ? (n--, i) : void 0, p && Ar(e[0], e[1], p) && (i = n < 3 ? void 0 : i, n = 1), r = Object(r); ++o < n; ) {
      var a = e[o];
      a && t(r, a, o, i);
    }
    return r;
  });
}
var Cr = La;

// ../../node_modules/.pnpm/lodash-es@4.17.21/node_modules/lodash-es/mergeWith.js
var Da = Cr(function(t, r, e, o) {
  gr(t, r, e, o);
}), Fa = Da;
export {
  Fa as mergeWith
};
