// @gqty/utils/sha1.js
"use strict";
var W = Object.create;
var g = Object.defineProperty;
var k = Object.getOwnPropertyDescriptor;
var N = Object.getOwnPropertyNames;
var Z = Object.getPrototypeOf, _ = Object.prototype.hasOwnProperty;
var B = (o, e) => () => (e || o((e = { exports: {} }).exports, e), e.exports), D = (o, e) => {
  for (var r in e)
    g(o, r, { get: e[r], enumerable: !0 });
}, x = (o, e, r, n) => {
  if (e && typeof e == "object" || typeof e == "function")
    for (let t of N(e))
      !_.call(o, t) && t !== r && g(o, t, { get: () => e[t], enumerable: !(n = k(e, t)) || n.enumerable });
  return o;
};
var E = (o, e, r) => (r = o != null ? W(Z(o)) : {}, x(e || !o || !o.__esModule ? g(r, "default", { value: o, enumerable: !0 }) : r, o)), G = (o) => x(g({}, "__esModule", { value: !0 }), o);

// ../../node_modules/.pnpm/crypt@0.0.2/node_modules/crypt/crypt.js
var A = B((L, S) => {
  (function() {
    var o = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/", e = {
      rotl: function(r, n) {
        return r << n | r >>> 32 - n;
      },
      rotr: function(r, n) {
        return r << 32 - n | r >>> n;
      },
      endian: function(r) {
        if (r.constructor == Number)
          return e.rotl(r, 8) & 16711935 | e.rotl(r, 24) & 4278255360;
        for (var n = 0; n < r.length; n++)
          r[n] = e.endian(r[n]);
        return r;
      },
      randomBytes: function(r) {
        for (var n = []; r > 0; r--)
          n.push(Math.floor(Math.random() * 256));
        return n;
      },
      bytesToWords: function(r) {
        for (var n = [], t = 0, u = 0; t < r.length; t++, u += 8)
          n[u >>> 5] |= r[t] << 24 - u % 32;
        return n;
      },
      wordsToBytes: function(r) {
        for (var n = [], t = 0; t < r.length * 32; t += 8)
          n.push(r[t >>> 5] >>> 24 - t % 32 & 255);
        return n;
      },
      bytesToHex: function(r) {
        for (var n = [], t = 0; t < r.length; t++)
          n.push((r[t] >>> 4).toString(16)), n.push((r[t] & 15).toString(16));
        return n.join("");
      },
      hexToBytes: function(r) {
        for (var n = [], t = 0; t < r.length; t += 2)
          n.push(parseInt(r.substr(t, 2), 16));
        return n;
      },
      bytesToBase64: function(r) {
        for (var n = [], t = 0; t < r.length; t += 3)
          for (var u = r[t] << 16 | r[t + 1] << 8 | r[t + 2], f = 0; f < 4; f++)
            t * 8 + f * 6 <= r.length * 8 ? n.push(o.charAt(u >>> 6 * (3 - f) & 63)) : n.push("=");
        return n.join("");
      },
      base64ToBytes: function(r) {
        r = r.replace(/[^A-Z0-9+\/]/ig, "");
        for (var n = [], t = 0, u = 0; t < r.length; u = ++t % 4)
          u != 0 && n.push((o.indexOf(r.charAt(t - 1)) & Math.pow(2, -2 * u + 8) - 1) << u * 2 | o.indexOf(r.charAt(t)) >>> 6 - u * 2);
        return n;
      }
    };
    S.exports = e;
  })();
});

// ../../node_modules/.pnpm/charenc@0.0.2/node_modules/charenc/charenc.js
var d = B((P, H) => {
  var y = {
    utf8: {
      stringToBytes: function(o) {
        return y.bin.stringToBytes(unescape(encodeURIComponent(o)));
      },
      bytesToString: function(o) {
        return decodeURIComponent(escape(y.bin.bytesToString(o)));
      }
    },
    bin: {
      stringToBytes: function(o) {
        for (var e = [], r = 0; r < o.length; r++)
          e.push(o.charCodeAt(r) & 255);
        return e;
      },
      bytesToString: function(o) {
        for (var e = [], r = 0; r < o.length; r++)
          e.push(String.fromCharCode(o[r]));
        return e.join("");
      }
    }
  };
  H.exports = y;
});

// ../../node_modules/.pnpm/sha1@1.1.1/node_modules/sha1/sha1.js
var q = B((Q, C) => {
  (function() {
    var o = A(), e = d().utf8, r = d().bin, n = function(u) {
      u.constructor == String ? u = e.stringToBytes(u) : typeof Buffer < "u" && typeof Buffer.isBuffer == "function" && Buffer.isBuffer(u) ? u = Array.prototype.slice.call(u, 0) : Array.isArray(u) || (u = u.toString());
      var f = o.bytesToWords(u), l = u.length * 8, p = [], s = 1732584193, c = -271733879, i = -1732584194, h = 271733878, v = -1009589776;
      f[l >> 5] |= 128 << 24 - l % 32, f[(l + 64 >>> 9 << 4) + 15] = l;
      for (var T = 0; T < f.length; T += 16) {
        for (var M = s, w = c, z = i, O = h, R = v, a = 0; a < 80; a++) {
          if (a < 16)
            p[a] = f[T + a];
          else {
            var F = p[a - 3] ^ p[a - 8] ^ p[a - 14] ^ p[a - 16];
            p[a] = F << 1 | F >>> 31;
          }
          var U = (s << 5 | s >>> 27) + v + (p[a] >>> 0) + (a < 20 ? (c & i | ~c & h) + 1518500249 : a < 40 ? (c ^ i ^ h) + 1859775393 : a < 60 ? (c & i | c & h | i & h) - 1894007588 : (c ^ i ^ h) - 899497514);
          v = h, h = i, i = c << 30 | c >>> 2, c = s, s = U;
        }
        s += M, c += w, i += z, h += O, v += R;
      }
      return [s, c, i, h, v];
    }, t = function(u, f) {
      var l = o.wordsToBytes(n(u));
      return f && f.asBytes ? l : f && f.asString ? r.bytesToString(l) : o.bytesToHex(l);
    };
    t._blocksize = 16, t._digestsize = 20, C.exports = t;
  })();
});

// lib/sha1.ts
var J = {};
D(J, {
  sha1: () => I.default
});
module.exports = G(J);
var I = E(q());
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  sha1
});
