// @gqty/utils/sha1.mjs
"use strict";
var R = Object.create;
var F = Object.defineProperty;
var U = Object.getOwnPropertyDescriptor;
var W = Object.getOwnPropertyNames;
var k = Object.getPrototypeOf, N = Object.prototype.hasOwnProperty;
var T = (o, e) => () => (e || o((e = { exports: {} }).exports, e), e.exports);
var Z = (o, e, r, n) => {
  if (e && typeof e == "object" || typeof e == "function")
    for (let t of W(e))
      !N.call(o, t) && t !== r && F(o, t, { get: () => e[t], enumerable: !(n = U(e, t)) || n.enumerable });
  return o;
};
var _ = (o, e, r) => (r = o != null ? R(k(o)) : {}, Z(e || !o || !o.__esModule ? F(r, "default", { value: o, enumerable: !0 }) : r, o));

// ../../node_modules/.pnpm/crypt@0.0.2/node_modules/crypt/crypt.js
var S = T((G, x) => {
  (function() {
    var o = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/", e = {
      rotl: function(r, n) {
        return r << n | r >>> 32 - n;
      },
      rotr: function(r, n) {
        return r << 32 - n | r >>> n;
      },
      endian: function(r) {
        if (r.constructor == Number)
          return e.rotl(r, 8) & 16711935 | e.rotl(r, 24) & 4278255360;
        for (var n = 0; n < r.length; n++)
          r[n] = e.endian(r[n]);
        return r;
      },
      randomBytes: function(r) {
        for (var n = []; r > 0; r--)
          n.push(Math.floor(Math.random() * 256));
        return n;
      },
      bytesToWords: function(r) {
        for (var n = [], t = 0, u = 0; t < r.length; t++, u += 8)
          n[u >>> 5] |= r[t] << 24 - u % 32;
        return n;
      },
      wordsToBytes: function(r) {
        for (var n = [], t = 0; t < r.length * 32; t += 8)
          n.push(r[t >>> 5] >>> 24 - t % 32 & 255);
        return n;
      },
      bytesToHex: function(r) {
        for (var n = [], t = 0; t < r.length; t++)
          n.push((r[t] >>> 4).toString(16)), n.push((r[t] & 15).toString(16));
        return n.join("");
      },
      hexToBytes: function(r) {
        for (var n = [], t = 0; t < r.length; t += 2)
          n.push(parseInt(r.substr(t, 2), 16));
        return n;
      },
      bytesToBase64: function(r) {
        for (var n = [], t = 0; t < r.length; t += 3)
          for (var u = r[t] << 16 | r[t + 1] << 8 | r[t + 2], f = 0; f < 4; f++)
            t * 8 + f * 6 <= r.length * 8 ? n.push(o.charAt(u >>> 6 * (3 - f) & 63)) : n.push("=");
        return n.join("");
      },
      base64ToBytes: function(r) {
        r = r.replace(/[^A-Z0-9+\/]/ig, "");
        for (var n = [], t = 0, u = 0; t < r.length; u = ++t % 4)
          u != 0 && n.push((o.indexOf(r.charAt(t - 1)) & Math.pow(2, -2 * u + 8) - 1) << u * 2 | o.indexOf(r.charAt(t)) >>> 6 - u * 2);
        return n;
      }
    };
    x.exports = e;
  })();
});

// ../../node_modules/.pnpm/charenc@0.0.2/node_modules/charenc/charenc.js
var y = T((J, A) => {
  var B = {
    utf8: {
      stringToBytes: function(o) {
        return B.bin.stringToBytes(unescape(encodeURIComponent(o)));
      },
      bytesToString: function(o) {
        return decodeURIComponent(escape(B.bin.bytesToString(o)));
      }
    },
    bin: {
      stringToBytes: function(o) {
        for (var e = [], r = 0; r < o.length; r++)
          e.push(o.charCodeAt(r) & 255);
        return e;
      },
      bytesToString: function(o) {
        for (var e = [], r = 0; r < o.length; r++)
          e.push(String.fromCharCode(o[r]));
        return e.join("");
      }
    }
  };
  A.exports = B;
});

// ../../node_modules/.pnpm/sha1@1.1.1/node_modules/sha1/sha1.js
var C = T((K, H) => {
  (function() {
    var o = S(), e = y().utf8, r = y().bin, n = function(u) {
      u.constructor == String ? u = e.stringToBytes(u) : typeof Buffer < "u" && typeof Buffer.isBuffer == "function" && Buffer.isBuffer(u) ? u = Array.prototype.slice.call(u, 0) : Array.isArray(u) || (u = u.toString());
      var f = o.bytesToWords(u), l = u.length * 8, p = [], s = 1732584193, c = -271733879, i = -1732584194, h = 271733878, v = -1009589776;
      f[l >> 5] |= 128 << 24 - l % 32, f[(l + 64 >>> 9 << 4) + 15] = l;
      for (var g = 0; g < f.length; g += 16) {
        for (var q = s, I = c, M = i, w = h, z = v, a = 0; a < 80; a++) {
          if (a < 16)
            p[a] = f[g + a];
          else {
            var d = p[a - 3] ^ p[a - 8] ^ p[a - 14] ^ p[a - 16];
            p[a] = d << 1 | d >>> 31;
          }
          var O = (s << 5 | s >>> 27) + v + (p[a] >>> 0) + (a < 20 ? (c & i | ~c & h) + 1518500249 : a < 40 ? (c ^ i ^ h) + 1859775393 : a < 60 ? (c & i | c & h | i & h) - 1894007588 : (c ^ i ^ h) - 899497514);
          v = h, h = i, i = c << 30 | c >>> 2, c = s, s = O;
        }
        s += q, c += I, i += M, h += w, v += z;
      }
      return [s, c, i, h, v];
    }, t = function(u, f) {
      var l = o.wordsToBytes(n(u));
      return f && f.asBytes ? l : f && f.asString ? r.bytesToString(l) : o.bytesToHex(l);
    };
    t._blocksize = 16, t._digestsize = 20, H.exports = t;
  })();
});

// lib/sha1.ts
var D = _(C());
var export_sha1 = D.default;
export {
  export_sha1 as sha1
};
