// @faustjs/next/dist/mjs/utils/assert.js
import isArray from 'lodash/isArray.js';
import isObject from 'lodash/isObject.js';
import isString from 'lodash/isString.js';
import trim from 'lodash/trim.js';
function hasPropString(prop) {
    return (params) => {
        return isObject(params) && isString(params[prop]);
    };
}
function hasPropStringArray(prop) {
    return (params) => {
        var _a;
        return (isObject(params) && isArray(params[prop]) && isString((_a = params[prop]) === null || _a === void 0 ? void 0 : _a[0]));
    };
}
export const hasPostId = hasPropString('postId');
export const hasPostSlug = hasPropString('postSlug');
export const hasPostUri = hasPropStringArray('postUri');
export const hasPageId = hasPropString('pageId');
export const hasPageUri = hasPropStringArray('pageUri');
export const hasCategoryId = hasPropString('categoryId');
export const hasCategorySlug = hasPropString('categorySlug');
export function formatUrlPrefix(pathPrefix) {
    if (pathPrefix.length === 1 && pathPrefix === '/') {
        return '';
    }
    const prefix = trim(pathPrefix, '/');
    return `/${prefix}`;
}
